lib/private/AppConfig.php

```diff
@@ -30,413 +33,1378 @@
 	 *
-	 * This function gets all keys of an app. Please note that the values are
-	 * not returned.
+	 * @return array<string, string> [appId => configValue]
+	 * @since 29.0.0
 	 */
-	public function getKeys($app) {
-		$this->loadConfigValues();
+	public function searchValues(string $key, bool $lazy = false): array {
+		$this->assertParams('', $key, true);
+		$this->loadConfig($lazy);
+		$values = [];
+
+		/** @var array<array-key, array<array-key, mixed>> $cache */
+		if ($lazy) {
+			$cache = &$this->lazyCache;
+		} else {
+			$cache = &$this->fastCache;
+		}
 
-		if (isset($this->cache[$app])) {
-			return $this->getSortedKeys($this->cache[$app]);
+		foreach (array_keys($cache) as $app) {
+			if (isset($cache[$app][$key])) {
+				$values[$app] = $cache[$app][$key];
+			}
 		}
 
-		return [];
+		return $values;
 	}
 
-	public function getSortedKeys($data) {
-		$keys = array_keys($data);
-		sort($keys);
-		return $keys;
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param string $default default value
+	 * @param bool $lazy search within lazy loaded config
+	 *
+	 * @return string stored config value or $default if not set in database
+	 * @throws InvalidArgumentException if one of the argument format is invalid
+	 * @throws AppConfigTypeConflictException in case of conflict with the value type set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy loading
+	 * @see getValueInt()
+	 * @see getValueFloat()
+	 * @see getValueBool()
+	 * @see getValueArray()
+	 */
+	public function getValueString(
+		string $app,
+		string $key,
+		string $default = '',
+		bool $lazy = false
+	): string {
+		return $this->getTypedValue($app, $key, $default, $lazy, self::VALUE_STRING);
 	}
 
 	/**
-	 * Gets the config value
+	 * @inheritDoc
 	 *
-	 * @param string $app app
-	 * @param string $key key
-	 * @param string $default = null, default value if the key does not exist
-	 * @return string the value or $default
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param int $default default value
+	 * @param bool $lazy search within lazy loaded config
 	 *
-	 * This function gets a value from the appconfig table. If the key does
-	 * not exist the default value will be returned
+	 * @return int stored config value or $default if not set in database
+	 * @throws InvalidArgumentException if one of the argument format is invalid
+	 * @throws AppConfigTypeConflictException in case of conflict with the value type set in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy loading
+	 * @see getValueString()
+	 * @see getValueFloat()
+	 * @see getValueBool()
+	 * @see getValueArray()
 	 */
-	public function getValue($app, $key, $default = null) {
-		$this->loadConfigValues();
+	public function getValueInt(
+		string $app,
+		string $key,
+		int $default = 0,
+		bool $lazy = false
+	): int {
```

Comment:
```suggestion
	): int|float {
```
If you want bigint support on 32bit